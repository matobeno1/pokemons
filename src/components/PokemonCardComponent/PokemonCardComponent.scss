@import "styles/mixins";

.pokemon-card-component {
  padding: var(--spacing-1);
  padding-top: var(--spacing-3);
  padding-bottom: var(--spacing-2);
  border-radius: var(--border-radius);
  display: flex;
  flex-direction: column;
  position: relative;
  cursor: pointer;

  @include transition(box-shadow, 200ms);
  @include box-shadow(2);

  &:hover, &:focus {
    @include box-shadow(6);
  }

  &:focus {
    outline: none;
  }

  &__backdrop {
    filter: brightness(150%) opacity(50%);
    border-radius: var(--border-radius);
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    top: 0;
    z-index: -1;
  }

  &__title {
    --title-size: var(--font-size-xs);
    color: var(--color-text);
    font-weight: 600;
    font-size: var(--title-size);
    text-align: center;
    padding-bottom: var(--spacing-3);
    display: block;
    text-transform: capitalize;

    @include md() {
      --title-size: var(--font-size-md);
    }

    @include lg() {
      --title-size: var(--font-size-lg);
    }
  }

  &__image {
    padding: var(--spacing-1);
  }
}
