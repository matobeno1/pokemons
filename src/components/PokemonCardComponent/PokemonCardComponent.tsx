import React, { FunctionComponent } from "react";
import { Pokemon } from "../../types";

type PokemonCardComponentProps = {
    pokemon: Pokemon
};

export const PokemonCardComponent: FunctionComponent<PokemonCardComponentProps> = ({
    pokemon
}) => {
    return (
        <div>

        </div>
    );
};
