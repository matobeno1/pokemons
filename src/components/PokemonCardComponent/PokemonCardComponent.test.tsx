import { act, create } from "react-test-renderer";
import { PokemonCardComponent } from "./PokemonCardComponent";

describe("PokemonCardComponent", () => {
    it("snapshots", async () => {
        let root: ReturnType<typeof create> | undefined;
        const name = "bulbasaur";
        const href = `/pokemons/${name}`;
        act(() => {
            root = create(
                <PokemonCardComponent
                    name={name}
                    href={href}
                />
            );
        });
        expect(root?.toJSON()).toMatchSnapshot();
    });
});
