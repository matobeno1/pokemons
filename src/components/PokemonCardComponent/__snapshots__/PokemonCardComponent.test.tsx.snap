// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCardComponent snapshots 1`] = `
<link
  href="/pokemons/bulbasaur"
  passHref={true}
>
  <a
    className="pokemon-card-component"
    role="link"
    tabIndex={0}
  >
    <div
      className="pokemon-card-component__backdrop"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <span
      className="pokemon-card-component__title"
    >
      bulbasaur
    </span>
    <div
      className="pokemon-card-component__image"
    />
  </a>
</link>
`;
